dsc_lib/src/discovery/command_discovery.rs

```diff
@@ -78,74 +62,241 @@ impl CommandDiscovery {
+                                        adapter_resources.insert(resource.type_name.to_lowercase(),resource.clone());
                                     }
-                                    continue;
-                                },
-                            };
-
-                            if resource.manifest.is_some() {
-                                let manifest = import_manifest(resource.manifest.clone().unwrap())?;
-                                if manifest.adapter.is_some() {
-                                    adapter_resources.push(resource.type_name.to_lowercase());
-                                    resources.insert(resource.type_name.to_lowercase(), resource.clone());
+                                }
+
+                                if type_regex.is_match(&resource.type_name) {
+                                    resources.insert(resource.type_name.to_lowercase(), resource);
                                 }
                             }
-                            if return_all_resources
-                            {
-                                resources.insert(resource.type_name.to_lowercase(), resource);
+                        }
+                    }
+                }
+            }
+        }
+        debug!("Found {} matching non-adapter-based resources", resources.len());
+
+        if !adapter_name_filter.is_empty() {
+            let regex_str = convert_wildcard_to_regex(adapter_name_filter);
+            debug!("Using regex {regex_str} as filter for adapter name");
+            let mut regex_builder = RegexBuilder::new(regex_str.as_str());
+            regex_builder.case_insensitive(true);
+            let adapter_regex: Regex;
+            if let Ok(reg_v) = regex_builder.build() {
+                adapter_regex = reg_v;
+            } else {
+                let err_str = "Could not build Regex filter for adapter name";
+                error!(err_str);
+                return Err(DscError::Operation(err_str.to_string()));
+            }
+
+            // now go through the adapter resources and add them to the list of resources
+            for adapter in adapter_resources {
+                if adapter_regex.is_match(&adapter.1.type_name) {
+                    debug!("Enumerating resources for adapter {}", adapter.1.type_name);
+                    let pb_adapter_span = warn_span!("");
+                    pb_adapter_span.pb_set_style(&ProgressStyle::with_template(
+                        "{spinner:.green} [{elapsed_precise:.cyan}] {msg:.white}"
+                    )?);
+                    pb_adapter_span.pb_set_message(format!("Enumerating resources for adapter {}", adapter.1.type_name).as_str());
+                    let _ = pb_adapter_span.enter();
+                    let adapter_resource = adapter.1;
+                    let adapter_type_name = adapter_resource.type_name.clone();
+                    let manifest = import_manifest(adapter_resource.manifest.clone().unwrap())?;
```

Comment:
Again, rethink all of your uses of `unwrap()`.  This can be:

```suggestion
                    let manifest = if let Some(manifest) = adapter_resource.manifest {
                      import_manifest(manifest)
                    } else {
                      return Err(DscError::MissingManifest(adapter_resource.type_name.clone()));
                    };
```

Reply:
updated.